scripts/lib/CIME/utils.py

```diff
@@ -1589,3 +1589,213 @@ def run_bld_cmd_ensure_logging(cmd, arg_logger, from_dir=None):
 
 def get_batch_script_for_job(job):
     return job if "st_archive" in job else "." + job
+
+class date:
```

Comment:
I feel like this class warrants its own file.

Reply:
I don't disagree, though I'm a little wary of cluttering the scripts directory with a utility class. If you're okay with it for this, I've moved it to a file just called `date.py`.

Reply:
Thanks.